modules/ws/src/test/java/org/mule/module/ws/functional/MtomFunctionalTestCase.java

```diff
@@ -123,5 +132,36 @@ private String extractAttachmentId(String payload)
         throw new IllegalArgumentException("Payload does not contain an attachment id");
     }
 
+    private void testAttachments(MuleMessage message) {
+        assertThat(message.getInvocationProperty("cxf_attachments"), notNullValue());
+
+        Collection<AttachmentImpl> attachments = message.getInvocationProperty("cxf_attachments");
+        assertThat(attachments, hasSize(1));
+        assertThat(attachments, hasItem(new AttachmentMatcher()));
+    }
+
+    private class AttachmentMatcher implements org.hamcrest.Matcher<AttachmentImpl> {
+
+        @Override
+        public boolean matches(Object o) {
+            AttachmentImpl attachment = (AttachmentImpl) o;
+            return attachment.getHeader(CONTENT_DISPOSITION).contains("attachment");
+        }
+
+        @Override
+        public void describeMismatch(Object o, Description description) {
+            description.appendText("Attachment doesn't have Content-Disposition Header, or it's not defined with the value 'attachment'");
+        }
+
+        @Override
+        public void _dont_implement_Matcher___instead_extend_BaseMatcher_() {
```

Comment:
what this line says :)

Reply:
https://www.youtube.com/watch?v=Ya9sjHUoBs8